src/h5web/visualizations/shared/AxisSystem.tsx

```diff
@@ -1,52 +1,49 @@
-import React, { useState } from 'react';
+import React, { useState, useContext } from 'react';
 import { useThree, useFrame, Dom } from 'react-three-fiber';
 import { AxisLeft, AxisBottom } from '@vx/axis';
 import { format } from 'd3-format';
 import { TickRendererProps } from '@vx/axis/lib/types';
 import { Grid } from '@vx/grid';
-import { scaleLinear } from 'd3-scale';
 import styles from './AxisSystem.module.css';
-import { AxisOffsets, AxisDomains } from './models';
-import { adaptedNumTicks, sceneToAxisScales } from './utils';
+import { AxisOffsets, AxisDomains, DataScale } from './models';
+import { adaptedNumTicks, useAbscissaScale, useOrdinateScale } from './utils';
+import { AxisSystemContext } from './AxisSystemProvider';
 
 interface Props {
   axisDomains: AxisDomains;
   axisOffsets: AxisOffsets;
-  showGrid?: boolean;
 }
 
 function AxisSystem(props: Props): JSX.Element {
-  const { axisDomains, axisOffsets, showGrid } = props;
+  const { axisDomains, axisOffsets } = props;
 
   const { camera, size } = useThree();
   const { position, zoom } = camera;
   const { width, height } = size;
 
-  const [domains, setDomains] = useState<AxisDomains>({
+  const { showGrid } = useContext(AxisSystemContext);
+
+  // axisDomains are the complete domains. visibleDomains change with the camera
+  const [visibleDomains, setVisibleDomains] = useState<AxisDomains>({
     left: axisDomains.left,
     bottom: axisDomains.bottom,
   });
 
-  const sceneToAxis = sceneToAxisScales(size, axisDomains);
-
-  const axisScales = {
-    x: scaleLinear()
-      .domain(domains.bottom)
-      .range([0, width]),
-    y: scaleLinear()
-      .domain(domains.left)
-      .range([height, 0]),
+  const axisToCanvas = {
+    x: useAbscissaScale(axisDomains.bottom) as DataScale,
+    y: useOrdinateScale(axisDomains.left) as DataScale,
```

Comment:
Why not merge the two hooks into one and return and return an object with `x` and `y` properties for consistency?

Reply:
Maybe not, I don't know. I just feel like we need to pick whether we use objects with `x` and `y` or individual variables and do so everywhere.